test/parallel/test-icu-locale.js

```diff
@@ -0,0 +1,203 @@
+    process.execPath,
+    [
+      '--icu-locale=fr-FR',
+      '-p',
+      'Intl?.Collator().resolvedOptions().locale',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'fr-FR\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=en-GB',
+      '-p',
+      'Intl?.Collator().resolvedOptions().locale',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'en-GB\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=en_GB',
+      '-p',
+      'Intl?.Collator().resolvedOptions().locale',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'en-GB\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=en',
+      '-p',
+      'Intl?.Collator().resolvedOptions().locale',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'en\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=de-DE',
+      '-p',
+      'Intl?.Collator().resolvedOptions().locale',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'de-DE\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=invalid',
+      '-p',
+      'Intl?.Collator().resolvedOptions().locale',
+    ]);
+  assert.strictEqual(child.stdout.toString(), 'en-US\n');
+}
+
+// NumberFormat
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    ['--icu-locale=de-DE', '-p', '(123456.789).toLocaleString(undefined, { style: "currency", currency: "EUR" })']
+  );
+  assert.strictEqual(child.stdout.toString(), '123.456,79\xa0€\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=ja-JP',
+      '-p',
+      '(123456.789).toLocaleString(undefined, { style: "currency", currency: "JPY" })',
+    ]);
+  assert.strictEqual(child.stdout.toString(), '￥123,457\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=en-IN',
+      '-p',
+      '(123456.789).toLocaleString(undefined, { maximumSignificantDigits: 3  })',
+    ]);
+  assert.strictEqual(child.stdout.toString(), '1,23,000\n');
+}
+// DateTimeFormat
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=en-GB',
+      '-p',
+      'new Date(Date.UTC(2012, 11, 20, 3, 0, 0)).toLocaleString(undefined,  { timeZone: "UTC" })',
+    ]);
+  assert.strictEqual(child.stdout.toString(), '20/12/2012, 03:00:00\n');
+}
+
+{
+  const { child } = spawnSyncAndExitWithoutError(
+    process.execPath,
+    [
+      '--icu-locale=ko-KR',
+      '-p',
+      'new Date(Date.UTC(2012, 11, 20, 3, 0, 0)).toLocaleString(undefined,  { timeZone: "UTC" })',
+    ]);
+  assert.strictEqual(child.stdout.toString(), '2012. 12. 20. 오전 3:00:00\n');
```

Comment:
I really don't like these kinds of tests. At all.  You are hard coding linguistic data that is guaranteed to change with every release, and even with local modifications the user makes.

I would recommend NOT testing for unsupported locales (you can't predict which locales are included), and not testing for specific formats.  We've discussed this elsewhere.

Reply:
I took these examples from MDN.

Reply:
You can't expect it. The data changes every single time.  See https://www.unicode.org/cldr/charts/latest/delta/index.html and click on any locale. 

I think a better test would simply be to verify that the output is DIFFERENT when a different locale is chosen.

Reply:
right. that is also a good approach.